synapse/storage/databases/state/bg_updates.py

```diff
@@ -154,7 +169,7 @@ def _get_state_groups_from_groups_txn(
                         f"""
                         (
                             SELECT DISTINCT ON (type, state_key)
-                                type, state_key, event_id
+                                type, state_key, event_id, state_group
                             FROM state_groups_state
                             INNER JOIN sgs USING (state_group)
```

Comment:
I don't think this `INNER JOIN` is necessary. Doesn't seem to be used :thinking:

Reply:
This is necessary because it will only return rows when there is a match from both tables.

It's unclear why we do this vs `WHERE state_group IN (SELECT state_group FROM sgs)` which we do in the non-optimized case.

Reply:
I think the query planner is smart enough to do the same thing in both cases :shrug: